tests/unit/template_cassdc_test.go

```diff
@@ -1177,19 +1202,19 @@ var _ = Describe("Verify CassandraDatacenter template", func() {
 
 		It("using 3.11.9 and a custom image", func() {
 			version := "3.11.9"
-			image := "my_cassandra:latest"
+			repository := "my_cassandra"
 			options := &helm.Options{
 				KubectlOptions: defaultKubeCtlOptions,
 				SetValues: map[string]string{
-					"cassandra.version": version,
-					"cassandra.image":   image,
+					"cassandra.version":          version,
+					"cassandra.image.repository": repository,
 				},
 			}
 
 			Expect(renderTemplate(options)).To(Succeed())
 
 			Expect(cassdc.Spec.ServerVersion).To(Equal(version))
-			Expect(cassdc.Spec.ServerImage).To(Equal(image))
+			Expect(cassdc.Spec.ServerImage).To(Equal("docker.io/" + repository + ":latest"))
```

Comment:
If we change the default registry to something like `quay.io` won't this test start breaking? Should we explicitly set the registry for this test?

Reply:
If we change the default registry then yes tests will need to be updated. I can clean up the tests a bit though by introducing a variable for the default registry. I am actually 👎 on explicitly setting the registry because the templates assume a default of `docker.io`.

Reply:
I'm with you on the :-1: on explicitly setting the registry. Maybe it's ok for the tests to break if we change the default.